### PR TITLE
Crystal: Add support for nightly builds

### PR DESCRIPTION
**Pull request by will**
@jhass I noticed the recv-keys is the same on http://nightly.crystal-lang.org/, does this look correct?

### CODE REVIEW

**Comment by jhass** [comment]
It should be, got a signing subkey for the main key.

**Comment by jhass** [comment]
Should we perhaps explicitly validate for `default` (and maybe also allow `latest` or something)?

File: lib/travis/build/script/crystal.rb [hunk 1]
```diff
@@ -10,10 +10,16 @@ def configure
             sh.echo 'Installing Crystal', ansi: :yellow
 
             sh.cmd %q(sudo sh -c 'apt-key adv --keyserver keys.gnupg.net --recv-keys 09617FD37CC06B54')
-            sh.cmd %q(sudo sh -c 'echo "deb http://dist.crystal-lang.org/apt crystal main" > /etc/apt/sources.list.d/crystal.list')
-            sh.cmd %q(sudo sh -c 'apt-get update')
 
-            sh.cmd %q(sudo apt-get install crystal)
+            if config[:crystal] == 'nightly'
+              sh.cmd %q(sudo sh -c 'echo "deb http://crystal-dist.aeshna.de/apt crystal main" > /etc/apt/sources.list.d/crystal-nightly.list')
```
Comment (jhass) [inline lib/travis/build/script/crystal.rb]: This should actually use `nightly.crystal-lang.org`, gotta update the index, sorry.

**Comment by will** [comment]
@jhass yeah I like `latest` better. 

Does the main apt let you install old versions with like `apt-get install crystal-0.7.3`?

**Comment by jhass** [comment]
Good question, I don't think so.

File: spec/build/script/csharp_spec.rb [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
   describe 'version switching' do
     it 'throws a error with a invalid version' do
-      data[:config][:mono] = 'foo'
+      data[:config][:crystal] = 'foo'
```
Comment (jhass) [inline spec/build/script/csharp_spec.rb]: Typo.

**Comment by jhass** [comment]
LGTM (I hope :P) :+1:

**Pull request merged**